### PR TITLE
Added switch-to-branch command

### PR DESCRIPTION
**Pull request by vjFaLk**
- [x] Added command to switch frappe and erpnext to specified branch
- [x] Add functionality to switch branch for all installed apps
- [x] Add functionality to specify apps /app 
- [x] Test Case

### CODE REVIEW

File: bench/commands/update.py [hunk 1]
```diff
@@ -115,6 +115,15 @@ def restart_update(kwargs):
 	os.execv(sys.argv[0], sys.argv[:2] + args)
 
 
+@click.command('switch-to-branch')
+@click.argument('branch')
+@click.option('--upgrade',is_flag=True)
+def switch_to_branch(branch, upgrade=False):
+	"Switch frappe and erpnext to master branch"
```
Comment (anandpdoshi) [inline bench/commands/update.py]: `to specified branch`

**Comment by anandpdoshi** [comment]
Progress for `all installed apps`? You will have to check if that branch exists for that app

File: bench/app.py [hunk 2]
```diff
@@ -150,24 +150,36 @@ def switch_branch(branch, apps=None, bench='.', upgrade=False):
...
+		apps = [name for name in os.listdir(apps_dir)
+			if os.path.isdir(os.path.join(apps_dir, name))]
 		if branch=="v4.x.x":
 			apps.append('shopping_cart')
+	if 'frappe' in apps:
+		version_upgrade = is_version_upgrade(bench=bench, branch=branch)
+		if version_upgrade[0] and not upgrade:
+			raise MajorVersionUpgradeException("Switching to {0} will cause upgrade from {1} to {2}. Pass --upgrade to confirm".format(branch, version_upgrade[1], version_upgrade[2]), version_upgrade[1], version_upgrade[2])
+	else:
+		version_upgrade = (False,)
```
Comment (vjFaLk) [inline bench/app.py]: Feels like a weird solution, looking for a better one.

Basically, the current code will check if the branch that is passed exists for frappe or not, which is a part of checking if changing the branch will cause a major version change.

Comment (anandpdoshi) [inline bench/app.py]: It checks version change of only frappe. Ideally it should check major version change for all apps and ask to proceed

Comment (vjFaLk) [inline bench/app.py]: Problem with that is that, we wouldn't know the git remote name of any custom app. We can do this for frappe-only apps though.

Comment (anandpdoshi) [inline bench/app.py]: Assume `upstream`

File: bench/app.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ def is_version_upgrade(bench='.', branch=None):
 	upstream_version = get_upstream_version('frappe', bench=bench, branch=branch)
 
 	if not upstream_version:
-		raise Exception("Current branch of 'frappe' not in upstream")
+		raise Exception("Specified branch of 'frappe' not in upstream")
```
Comment (anandpdoshi) [inline bench/app.py]: `raise Exception("Branch '{0}' of 'frappe' not in upstream".format(branch))`

**Comment by anandpdoshi** [comment]
- [ ] Test Case

**Comment by anandpdoshi** [comment]
merged manually after fixing test case

**Pull request closed without merging**